### PR TITLE
add old mushroom hitbox

### PR DESCRIPTION
**Pull request by KhafraDev**
https://discord.com/channels/879732108745125969/1302395598586253343
https://discord.com/channels/879732108745125969/1167518820362100847

etc.

I have been using this for months in a private mod.

(this is untested in skyblocker, I ripped it straight from my mod pretty much and my java isn't the best)

### CODE REVIEW

**Comment by qtlunya** [comment]
If this gets added, can we add cactus too? That's also problematic on modern versions.

**Review by AzureAaron — CHANGES_REQUESTED** [review]

File: src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java
[the remark this answers is not in the document]
Comment (KhafraDev) [reply src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java:1]: MushroomPlantBlock doesn't have a getCullingShape method

File: src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	}
+
+	@ModifyReturnValue(method = "getOutlineShape", at = @At("RETURN"))
+	private VoxelShape skyblocker$getOldMushroomOutline(VoxelShape original) {
+		return Utils.isOnSkyblock() && SkyblockerConfigManager.get().general.hitbox.oldMushroomHitbox ? OLD_SHAPE : original;
+	}
+
+	@Override
+	public VoxelShape getCullingShape(BlockState state) {
+		return OLD_SHAPE;
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java:28]: I meant that you need to shadow in the original shape and return that

**Review by AzureAaron — CHANGES_REQUESTED** [review]

File: src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java (hunk 1 above)
Comment (KhafraDev) [reply src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java]: whoops

File: src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java [hunk 2]
```diff
@@ -6,11 +6,15 @@
...
 import org.spongepowered.asm.mixin.injection.At;
 
 @Mixin(MushroomPlantBlock.class)
-public class MushroomPlantBlockMixin extends Block {
+public abstract class MushroomPlantBlockMixin extends Block {
+	@Shadow
+	protected static VoxelShape SHAPE;
+
 	@Unique
 	private static final VoxelShape OLD_SHAPE = Block.createCuboidShape(4.8, 0.0, 4.8, 11.2, 6.4, 11.2);
```
Comment (viciscat) [inline src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java:19]: pretty sure this is now unused

Comment (KhafraDev) [reply src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java:19]: https://github.com/SkyblockerMod/Skyblocker/pull/1051/files#diff-2e45e569b555b94fdab5b3f4f7ea808eeaad80097053655aa8934da78f906c87R27

Comment (viciscat) [reply src/main/java/de/hysky/skyblocker/mixins/MushroomPlantBlockMixin.java:19]: oh right me dumb

**Review by AzureAaron — APPROVED** [review]
Didn't test this but it seems fine

**Comment by KhafraDev** [comment]
I've been testing it out for a few days and it works as expected

**Pull request merged**